File: features/lead/src/main/java/com/nivasafinance/features/lead/exception/LeadConflictException.java
package com.nivasafinance.features.lead.exception;

import com.nivasafinance.common.exception.ConflictException;
import com.nivasafinance.common.exception.ExceptionUtils;
import org.springframework.context.MessageSource;

public class LeadConflictException extends ConflictException {

    public LeadConflictException(String messageKey, Object[] args, MessageSource messageSource) {
        super(ExceptionUtils.INSTANCE.createLocalizedMessage(messageKey, args, messageSource));
    }
}

File: features/lead/src/main/java/com/nivasafinance/features/lead/exception/LeadConflictException.kt
package com.nivasafinance.features.lead.exception

import com.nivasafinance.common.exception.ConflictException
import com.nivasafinance.common.exception.ExceptionUtils
import org.springframework.context.MessageSource
import java.util.UUID

open class LeadConflictException(
    messageKey: String,
    args: Array<Any>? = null,
    messageSource: MessageSource
) : ConflictException(
    ExceptionUtils.createLocalizedMessage(messageKey, args, messageSource)
)

class LeadAlreadyExistsException(
    leadId: UUID,
    messageSource: MessageSource
) : LeadConflictException(
    "error.lead.already.exists",
    arrayOf(leadId.toString()),
    messageSource
)
